cli/localci/utils/yq.py

```
@@ -67,19 +68,56 @@ class YqWrapper:
+                "Install with: sudo snap install yq"
+            )
+
+    @staticmethod
+    def _detect_yq_flavour(yq_path: str) -> str:
+        """Return 'mikefarah', 'kislyuk', or 'unknown' based on --version output.
+
+        mikefarah/yq:  'yq (https://github.com/mikefarah/yq/) version v4.x.x'
+        kislyuk/yq:    'yq x.x.x' (no URL) — Python wrapper around jq
+        """
+        try:
+            result = subprocess.run(
```

Ruff (0.15.4)

[error] `subprocess` call: check for execution of untrusted input

(S603)